code/controllers/subsystem/weeds_decay.dm

```diff
@@ -0,0 +1,51 @@
+SUBSYSTEM_DEF(weeds_decay)
+	name = "Weed Decay"
+	priority = FIRE_PRIORITY_WEED
+	runlevels = RUNLEVEL_LOBBY|RUNLEVEL_SETUP|RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+	wait = 15 SECONDS
+
+	// This is a list of nodes on the map.
+	var/list/decaying = list()
+
+/datum/controller/subsystem/weeds_decay/stat_entry()
+	return ..("Decay Nodes: [length(decaying)]")
+
+/datum/controller/subsystem/weeds_decay/fire(resumed = FALSE)
+
+	for(var/A in decaying)
+		if(MC_TICK_CHECK)
+			return
+
+		var/turf/T = A
+		if(QDELETED(T))
+			decaying -= T
+			continue
+
+		var/obj/effect/alien/weeds/W = locate() in T
+		if(QDELETED(W) || W.parent_node)
+			decaying -= T
+			continue
+
+		var/decay_chance = 100
+		for(var/direction in GLOB.cardinals) 
+			var/turf/adj = get_step(T, direction)
+			if(locate(/obj/effect/alien/weeds) in adj)
+				decay_chance -= 25
```

Comment:
Maybe something between 20 and 24 so there's a small chance for weeds in the middle of a large pattern to die as well?